Source/cmMakefileTargetGenerator.cxx

```diff
@@ -1800,14 +1800,59 @@ void cmMakefileTargetGenerator::CloseFileStreams()
   this->FlagFileStream.reset();
 }
 
-void cmMakefileTargetGenerator::CreateLinkScript(
-  const char* name, std::vector<std::string> const& link_commands,
-  std::vector<std::string>& makefile_commands,
-  std::vector<std::string>& makefile_depends)
+//GLEB CHANGES
+void cmMakefileTargetGenerator::CreateLinkScriptJSON(const char *name, const std::vector<std::string>& files,
+                                                     const std::vector<std::string> &link_commands) {
+
+    // Create the link script file.
+    std::string linkScriptName =
+            cmStrCat(this->TargetBuildDirectoryFull, '/', name);
+
+    //GLEB CHANGES
+    std::string workingDirectory = cmSystemTools::CollapseFullPath(
+            this->LocalGenerator->GetCurrentBinaryDirectory());
+
+    std::vector<std::string> resultStringArr;
+    std::string delimiter = " ";
+    size_t pos = 0;
+    std::string token;
+    for (auto file : files){
```

Comment:
const &

Reply:
> const &

it's impossible in this case. I change value using erase method

Reply:
> const &

I cant use it. File is mutable variable